src/Tooltip/Tooltip.component.tsx

```diff
@@ -0,0 +1,262 @@
+                left: 0,
+                top: 0,
+                transform: `translate3d(${width / 2 -
+                    toolTipWidth / 2}px, ${height + tooltipMargin}px, 0)`,
+            });
+        case 'bottomStart':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(0, ${height + tooltipMargin}px, 0)`,
+            });
+        case 'leftEnd':
+            return css({
+                left: 0,
+                bottom: 0,
+                transform: `translate3d(-${toolTipWidth +
+                    tooltipMargin}px, 0, 0)`,
+            });
+        case 'left':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(-${toolTipWidth +
+                    tooltipMargin}px, ${height / 2 - toolTipHeight / 2}px, 0)`,
+            });
+        case 'leftStart':
+            return css({
+                left: 0,
+                top: 0,
+                transform: `translate3d(-${toolTipWidth +
+                    tooltipMargin}px, 0, 0)`,
+            });
+        default:
+            return css({
+                right: 0,
+                top: 0,
+                transform: `translate3d(0, -${height + tooltipMargin}px, 0)`,
+            });
+    }
+};
+
+const TooltipElement = styled('div')<TooltipElementProps>`
+    position: absolute;
+    box-sizing: border-box;
+    ${({ background = 'rgba(0, 0, 0, 0.8)', color = 'white' }) =>
+        css({ background, color })};
+    border-radius: base;
+    font-family: ${th('typography.fontFamily')};
+    font-size: 0.8rem;
+    padding: 0.5rem;
+    opacity: 0;
+    visibility: hidden;
+    transition: opacity 250ms ease-in-out;
+
+    &.active {
+        opacity: 1;
+        visibility: visible;
+    }
+
+    ${({ wrap = true }) =>
+        wrap ? css({ whiteSpace: 'normal' }) : css({ whiteSpace: 'nowrap' })};
+
+    ${({ maxWidth = 'auto' }) => css({ width: maxWidth })};
+
+    ${({ position, height, width, toolTipHeight, toolTipWidth }) =>
+        positionVariants(position, height, width, toolTipHeight, toolTipWidth)};
+`;
+
+export class Tooltip extends React.Component<
+    TooltipContainerProps,
+    TooltipContainerState
+> {
+    public state = {
+        width: 0,
+        height: 0,
+        toolTipHeight: 0,
+        toolTipWidth: 0,
+        hidden: true,
+    };
+    private readonly tooltipContainerRef: React.RefObject<
+        HTMLDivElement
+    > = React.createRef();
+    private readonly tooltipRef: React.RefObject<
+        HTMLDivElement
+    > = React.createRef();
+
+    componentDidMount(): void {
+        const { current: toolTipContainer } = this.tooltipContainerRef;
+        const { current: toolTip } = this.tooltipRef;
+        // TODO: instead of setting state, what about using the render fxn?
+        if (toolTipContainer) {
+            this.setState({
+                height: toolTipContainer.clientHeight,
```

Comment:
No listeners for resizing events or sommat?

Reply:
Nope. This isn't a dropdown; it's a tooltip.